talker/src/talker.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (c) 2019, ROSIN-project
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this
+list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+ * This tutorial demonstrates simple sending of messages over the ROS system 
+ * using the rxros interface
+ */
+
+#include <rxros/rxros.h>
+#include "std_msgs/String.h"
+
+using namespace rxcpp::operators;
+using namespace rxros::operators;
```

Comment:
I always tell my students I don't want to see any blanket `using namespace` statements in global scopes.

Reply:
Forgive me my ignorance about C++ (the previous time I used it professionally was before 1999).   What is the alternative?

Reply:
You're importing all symbols from `rxros::operators` and `rxcpp::operators` into the global scope here, which has a potential for clashing with other (external) symbols.

It's not good practice to do this.

>  What is the alternative?

Either move the `using` statement closer to where you actually need the symbols you're importing, or prefix everything just with `rxros::operators::`.

Reply:
I'll consider moving closer.  The other option, prefixing, is nonsense for short infix operators.  It destroys the point of using them.  The typechecker should detect an error if there is a clash (and then some measures can be taken).